src/map/status.c

```diff
@@ -1560,6 +1560,42 @@ static int status_fixed_revive(struct block_list *bl, unsigned int per_hp, unsig
 	return 1;
 }
 
+/**
+ * a sub-function from 'status_check_skilluse' function, which can be use for plugin manipulation
+ * check 'disabled_skills' flag settings from `db\(pre-)re\map_zone_db.conf`
+ * @param src        The source object can be BL_PC, BL_MOB, BL_PET, BL_HOM, BL_MER, BL_ELEM type
+ * @param st         The status_data of the source object, use to check MD_BOSS of BL_MOB type
+ * @param skill_id   The skill ID use
+ * @return false if the skill ID for the src is denied, true otherwise
+ **/
+static bool status_check_skilluse_mapzone(struct block_list *src, struct status_data *st, uint16 skill_id)
+{
+	nullpo_ret(src);
+	nullpo_ret(st);
```

Comment:
If you keep using `bool` as return type, use `nullpo_retr(false, src);`/`nullpo_retr(false, st);`.
If you change the return type to `int`, use `nullpo_retr(1, src);`/`nullpo_retr(1, st);`.

Reply:
```c
/**
 * Returns 0 if a NULL pointer is found.
 *
 * @param t pointer to check
 */
#define nullpo_ret(t) \
	do { if (nullpo_chk(t)) return(0); } while(0)
```
I think this is correct though ...

Reply:
here should be used ``nullpo_retr(false, ...)``
because ``nullpo_ret`` returning only 0. but 0 is not false. it same if cast to each other but not same without cast

Reply:
https://github.com/HerculesWS/Hercules/blob/0a01e23a2a37d86281d01a8638017e6e8b35d6c8/src/map/achievement.c#L149-L155

ok...